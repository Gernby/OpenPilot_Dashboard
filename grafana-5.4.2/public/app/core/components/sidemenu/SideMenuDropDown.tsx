import React, { SFC } from 'react';
import DropDownChild from './DropDownChild';

interface Props {
  link: any;
}

const SideMenuDropDown: SFC<Props> = props => {
  const { link } = props;
  return (
    <ul className="dropdown-menu dropdown-menu--sidemenu" role="menu">
      <li className="side-menu-header">
        <span className="sidemenu-item-text">{link.text}</span>
      </li>
      {link.children &&
        link.children.map((child, index) => {
          return <DropDownChild child={child} key={`${child.url}-${index}`} />;
        })}
    </ul>
  );
};

export default SideMenuDropDown;
