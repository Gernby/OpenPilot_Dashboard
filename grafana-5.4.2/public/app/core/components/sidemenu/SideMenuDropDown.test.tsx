import React from 'react';
import { shallow } from 'enzyme';
import SideMenuDropDown from './SideMenuDropDown';

const setup = (propOverrides?: object) => {
  const props = Object.assign(
    {
      link: {
        text: 'link',
      },
    },
    propOverrides
  );

  return shallow(<SideMenuDropDown {...props} />);
};

describe('Render', () => {
  it('should render component', () => {
    const wrapper = setup();

    expect(wrapper).toMatchSnapshot();
  });

  it('should render children', () => {
    const wrapper = setup({
      link: {
        text: 'link',
        children: [{ id: 1 }, { id: 2 }, { id: 3 }],
      },
    });

    expect(wrapper).toMatchSnapshot();
  });
});
