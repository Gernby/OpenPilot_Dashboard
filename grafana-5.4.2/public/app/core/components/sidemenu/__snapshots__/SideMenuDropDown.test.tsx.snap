// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render should render children 1`] = `
<ul
  className="dropdown-menu dropdown-menu--sidemenu"
  role="menu"
>
  <li
    className="side-menu-header"
  >
    <span
      className="sidemenu-item-text"
    >
      link
    </span>
  </li>
  <DropDownChild
    child={
      Object {
        "id": 1,
      }
    }
    key="undefined-0"
  />
  <DropDownChild
    child={
      Object {
        "id": 2,
      }
    }
    key="undefined-1"
  />
  <DropDownChild
    child={
      Object {
        "id": 3,
      }
    }
    key="undefined-2"
  />
</ul>
`;

exports[`Render should render component 1`] = `
<ul
  className="dropdown-menu dropdown-menu--sidemenu"
  role="menu"
>
  <li
    className="side-menu-header"
  >
    <span
      className="sidemenu-item-text"
    >
      link
    </span>
  </li>
</ul>
`;
